precice.f90

```diff
@@ -312,4 +314,21 @@ end subroutine precicef_get_version_information
 
   end interface
 
+  contains
```

Comment:
I still want that users call `precicef_create`, so they should be outside the `contains` block. However, it looks like I cannot call subroutines from inside an interface. What would be the right way here?

Reply:
But you have it now as a module procedure (a thin wrapper of the preCICE C API). Isn't that what you want?

Reply:
I want that users call the `precicef_create` variant without having to provide neither null-terminated strings, nor placeholder lengths. For this, I would need the `precicef_create` to not be internal (inside the `contains`), no?

If I ask the users to call the `precicec_create`, then they need to provide null-terminated strings.

Should I maybe split it into two different modules?

Reply:
In Fortran parlance, interface blocks are like function declarations in C. They are _specification_ statements (i.e. non-executable). 

When you want to _define_ a procedure, it must be in the "contains" part:
```
module precice
 ! ... specification ...
contains
 ! ... procedure definitions (including executable statements) ...
end module
```

Reply:
Ah, that helps! But then, I understand that both what I have in the interface and in the `contains` need to have the same name.